src/neuroconv/datainterfaces/text/timeintervalsinterface.py

```diff
@@ -0,0 +1,63 @@
+from abc import abstractmethod
+
+import numpy as np
+from pynwb import NWBFile
+from pynwb.epoch import TimeIntervals
+
+from ...basedatainterface import BaseDataInterface
+from ...utils.types import FilePathType, Optional
+from ...tools.nwb_helpers import make_or_load_nwbfile
+
+
+def convert_df_to_time_intervals(df, name, description=None):
+    if description is None:
+        description = name
+    time_intervals = TimeIntervals(name, description)
+    if "start_time" not in df:
+        raise ValueError(f"df must contain a column named 'start_time'. Existing columns: {df.columns.to_list()}")
+    if "stop_time" not in df:
+        df["stop_time"] = np.nan
+    for col in df:
+        if col not in ("start_time", "stop_time"):
+            time_intervals.add_column(col, col)
+    for i, row in df.iterrows():
+        time_intervals.add_row(row.to_dict())
+
+    return time_intervals
+
+
+class TimeIntervalsInterface(BaseDataInterface):
+    def __init__(
+        self,
+        file_path: FilePathType,
+        read_kwargs: Optional[dict] = None,
```

Comment:
Yeah, `read_kwargs` is probably the best way to get around complexity in reading strangely formatted CSV files (multiple tabs, several header offsets, fill values, etc.)

Reply:
yeah that's what I was thinking